source/transactions-convenient-api/tests/README.rst

```diff
@@ -28,16 +28,33 @@ Test Format
 
 Each YAML file has the following keys:
 
+- ``runOn`` (optional): An array of server version and/or topology requirements
+  for which the tests can be run. If the test environment satisfies one or more
+  of these requirements, the tests may be executed; otherwise, this file should
+  be skipped. If this field is omitted, the tests can be assumed to have no
+  particular requirements and should be executed. Each element will have some or
+  all of the following fields:
+
+  - ``minServerVersion`` (optional): The minimum server version (inclusive)
+    required to successfully run the tests. If this field is omitted, it should
+    be assumed that there is no lower bound on the required server version.
+
+  - ``maxServerVersion`` (optional): The maximum server version (exclusive)
```

Comment:
The maximum server version is inclusive in the existing tests, can it be also inclusive here?

Reply:
Where did you see an inclusive maximum server version? AFAIK, only the retryable writes and CRUD tests use `maxServerVersion` (apart from change streams, which says it is "reserved for future use") and they document it as an exclusive upper bound.

Reply:
I was referring to this: https://github.com/mongodb/specifications/tree/master/source/command-monitoring/tests#ignoring-tests-based-on-server-version

The current uses of maxserverversion are:

<pre>
spec/spec_tests/data/crud/write/findOneAndReplace-upsert_pre_2.6.yml:maxServerVersion: '2.4.99'
spec/spec_tests/data/crud/write/replaceOne-pre_2.6.yml:maxServerVersion: '2.4.99'
spec/spec_tests/data/crud/write/updateMany-pre_2.6.yml:maxServerVersion: '2.4.99'
spec/spec_tests/data/crud/write/updateOne-pre_2.6.yml:maxServerVersion: '2.4.99'
</pre>

These are all inclusive specifications.

Also, "max server version: 1.0" implies that 1.0 is an acceptable version just looking at it from English grammar standpoint.

Reply:
I will make this change for this PR and #510. I also mentioned this in https://jira.mongodb.org/browse/SPEC-1255, as that will pertain to changing the CRUD spec tests mentioned above.